util/types/json/json.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"bytes"
+
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/terror"
+)
+
+// JSON is for MySQL JSON type.
+type JSON interface {
```

Comment:
Got it. Can you use a JSONObject interface, Let Bool, Int***, map[string]JSONObject, []JSONObject implement this interface.
This interface contains GetType(), GetBytes(), etc...